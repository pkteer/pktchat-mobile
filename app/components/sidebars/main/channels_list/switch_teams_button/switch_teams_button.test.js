// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {shallow} from 'enzyme';
import React from 'react';

import Preferences from '@mm-redux/constants/preferences';

import SwitchTeamsButton from './switch_teams_button';

describe('SwitchTeamsButton', () => {
    const baseProps = {
        onShowTeams: jest.fn(),
        testID: 'main.sidebar.channels_list.switch_teams.button',
        currentTeamId: 'current-team-id',
        mentionCount: 1,
        teamsCount: 2,
        theme: Preferences.THEMES.denim,
    };

    test('should match snapshot', () => {
        const wrapper = shallow(<SwitchTeamsButton {...baseProps}/>);

        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
