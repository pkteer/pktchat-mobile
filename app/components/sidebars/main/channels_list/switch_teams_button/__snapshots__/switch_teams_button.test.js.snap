// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchTeamsButton should match snapshot 1`] = `
<View
  testID="main.sidebar.channels_list.switch_teams.button"
>
  <TouchableHighlight
    onPress={[Function]}
    underlayColor="rgba(25,42,77,0.5)"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(255,255,255,0.24)",
          "borderRadius": 2,
          "flexDirection": "row",
          "height": 36,
          "justifyContent": "center",
          "marginRight": 12,
          "width": 57,
        }
      }
      testID="main.sidebar.channels_list.switch_teams.button.current-team-id"
    >
      <CompassIcon
        name="chevron-left"
        size={24}
        style={
          Object {
            "color": "#ffffff",
            "left": -2,
            "top": 1,
          }
        }
      />
      <Connect(TeamIcon)
        styleContainer={
          Object {
            "alignItems": "center",
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          }
        }
        styleImage={
          Object {
            "height": 24,
            "width": 24,
          }
        }
        styleText={
          Object {
            "fontFamily": "Open Sans",
            "fontSize": 14,
          }
        }
        teamId="current-team-id"
        testID="main.sidebar.channels_list.switch_teams.button.team_icon"
      />
    </View>
  </TouchableHighlight>
  <Badge
    containerStyle={
      Object {
        "borderColor": "#1e325c",
        "borderRadius": 14,
        "borderWidth": 2,
        "position": "absolute",
        "right": 0,
        "top": -9,
      }
    }
    count={1}
    countStyle={
      Object {
        "color": "#1e325c",
        "fontSize": 10,
        "fontWeight": "bold",
      }
    }
    extraPaddingHorizontal={10}
    minHeight={20}
    minWidth={20}
    style={
      Object {
        "backgroundColor": "#ffffff",
        "height": 20,
        "padding": 3,
      }
    }
    testID="main.sidebar.channels_list.switch_teams.button.badge"
  />
</View>
`;
